misc/formatter/modules/formatter-cli/src/test/java/org/ballerinalang/formatter/cli/FormatCmdTest.java

```diff
@@ -82,6 +87,30 @@ public void formatCLIOnBallerinaProject() {
         }
     }
 
+    @Test(description = "Test ballerina project formatting")
+    public void formatCLIOnBallerinaProjectWithConfigurations() {
+        List<String> argList = new ArrayList<>();
+        for (Path dir : BALLERINA_CONFIGURATION_DIRS) {
+            try {
+                List<Path> subDirs = Files.list(dir)
+                        .filter(Files::isDirectory)
+                        .filter(d -> !d.getFileName().toString().endsWith("Temp"))
+                        .collect(Collectors.toList());
+                for (Path subDir : subDirs) {
+                    Path tempDir = subDir.resolveSibling(subDir.getFileName() + "Temp");
+                    Path assertDir = Paths.get(subDir.toString().replace("/source/", "/assert/"));
+
+                    FormatUtil.execute(argList, false, null, null, false, subDir);
+                    Assert.assertEquals(Files.readString(subDir.resolve("main.bal")),
+                            Files.readString(assertDir.resolve("main.bal")));
+                    FileUtils.copyDirectory(tempDir.toFile(), subDir.toFile());
+                }
+            } catch (IOException e) {
+                Assert.fail("Failed to update the source file.");
+            }
+        }
```

Comment:
IMO, it would be better if we could parameterize this test case. This way, if a test fails, we can precisely determine which one. Refer the usage of the data provider test case: https://github.com/nipunayf/ballerina-lang/blob/20bef5c3fe604ee1efe7fe298e860a74c97d76af/compiler/ballerina-lang/src/test/java/io/ballerina/projects/CyclicDependenciesTest.java#L61-L73